Modules/Sources/WordPressKit/ReaderFeed.swift

```diff
@@ -30,61 +66,52 @@ public struct ReaderFeed: Decodable {
     }
 
     public init(from decoder: Decoder) throws {
-        // We have to manually decode the feed from the JSON, for a couple of reasons:
-        // - Some feeds have no `icon` dictionary
-        // - Some feeds have no `data` dictionary
-        // - We want to decode whatever we can get, and not fail if neither of those exist
-        let rootContainer = try decoder.container(keyedBy: CodingKeys.self)
-
-        var feedURL = try? rootContainer.decodeIfPresent(URL.self, forKey: .url)
-        var title = try? rootContainer.decodeIfPresent(String.self, forKey: .title)
-        feedID = try? rootContainer.decode(String.self, forKey: .feedID)
-        blogID = try? rootContainer.decode(String.self, forKey: .blogID)
-
-        var feedDescription: String?
-        var blavatarURL: URL?
-
-        // Try to parse both site and feed data from meta.data
-        do {
-            let metaContainer = try rootContainer.nestedContainer(keyedBy: MetaKeys.self, forKey: .meta)
-            let dataContainer = try metaContainer.nestedContainer(keyedBy: DataKeys.self, forKey: .data)
-
-            let siteData = try? dataContainer.decode(SiteOrFeedData.self, forKey: .site)
-            let feedData = try? dataContainer.decode(SiteOrFeedData.self, forKey: .feed)
-
-            // Use data from either source, preferring site data when both are available
-            feedDescription = siteData?.description ?? feedData?.description
-            blavatarURL = siteData?.iconURL ?? feedData?.iconURL
-
-            // Fixes CMM-1002: in some cases, the backend fails to embed certain fields
-            // directly in the feed object
-            if feedURL == nil {
-                feedURL = siteData?.url ?? feedData?.url
-            }
-            if title == nil {
-                title = siteData?.title ?? feedData?.title
-            }
-        } catch {
+        let root = try decoder.container(keyedBy: CodingKeys.self)
+        if let meta = try? root.nestedContainer(keyedBy: MetaKeys.self, forKey: .meta),
+           let data = try? meta.nestedContainer(keyedBy: DataKeys.self, forKey: .data) {
+            self.feed = try? data.decode(FeedData.self, forKey: .feed)
+            self.site = try? data.decode(SiteData.self, forKey: .site)
         }
```

Comment:
I feel like manually decoding defeats the purpose of using `Decodable`. The code would look cleaner if we use types that match the JSON structure, like

```swift
public struct ReaderFeed: Decodable {
    public init(from decoder: Decoder) throws {
        let feed = try Feed(from: decoder)
        self.feed = feed.meta?.data?.feed
        self.site = feed.meta?.data?.site
    }
}

// For parsing JSON only.
private struct Feed: Decodable {
    struct Meta: Decodable {
        struct Data: Decodable {
            var feed: FeedData?
            var site: SiteData?
        }

        var data: Data?
    }

    var meta: Meta?
}
```

Reply:
That makes sense. I added `ReaderFeedJSON` just like that. It's a bit easier to follow now.